apis/v1alpha1/httproute_types.go

```diff
@@ -139,6 +140,8 @@ type HTTPRouteMatch struct {
 	// Default: "Exact"
 	//
 	// +optional
+	// +kubebuilder:validation:Enum=Exact;ImplementationSpecific
+	// +kubebuilder:default=Exact
```

Comment:
Adding a Go type can make this clearer for API users:

```Go
// Pretty sure that the validation is transitive here ...
// +kubebuilder:validation:Enum=Exact;ImplementationSpecific
type HeaderMatchType string

// We already have these constants, just need to add types.
const (
ImplementationSpecificHeaderMatch HeaderMatchType = "ImplementationSpecific"
ExactHeaderMatch HeaderMatchType = "Exact"
)

type HTTPRouteMatch struct {
...
// +kubebuilder:default=Exact
HeaderMatchType *string `json:"headerMatchType" protobuf:"bytes,3,opt,name=headerMatchType"`
}

```

Reply:
Can we do this in a separate PR?